### ISSUE TITLE
Why use post-command-hook?

### ISSUE THREAD

**Issue by raxod502** [10sr/switch-buffer-functions-el#1]
It was pointed out [here](https://github.com/flycheck/flycheck/pull/1308#issuecomment-325122360) that you might be able to achieve the same effect of `switch-buffer-functions.el` using `buffer-list-update-hook` instead, which would be less wasteful. Is that the case?

**Comment by 10sr** [comment]
Thanks!

I just did not know about `buffer-list-update-hook`...
Using `post-command-hook` is a really easy way as you know, and in practice it did not cause significant overhead with machines I have, so I chose this hook.
However, I agree it is much nicer if I can switch to use hook other than `post-command-hook`.
I'll try to implement with `buffer-list-update-hook` shortly!

(It is possible that using `buffer-list-update-hook` can cause infinite loop when badly implemented, but I think there should be a way to avoid it...)
